### DISCUSSION TITLE
How to display oc wires in Threejs?

### DISCUSSION THREAD

**Discussion by koodistrom** [donalffons/opencascade.js#260] [Q&A] [answered]
Hi, 
and thanks for an awesome open source library.  I am using opencascade.js to build programmatic cad tool with web preview of the object being created. I am using Threejs to display the objects.
Displaying the solid 3D shapes is easy. I write them as glxf and load them to threejs scene.

How ever I haven't found any easy way to  display wires. I would like to display them as lines in the Threejs view. Any ideas how this could be donet?

**Comment by frading** [comment, marked as answer]
Here is how I do it in [Polygonjs](https://github.com/polygonjs/polygonjs), if that helps:

- 1. traverse the edges of the wire ( https://github.com/polygonjs/polygonjs/blob/master/src/core/geometry/modules/cad/toObject3D/CadWire.ts#L18 and https://github.com/polygonjs/polygonjs/blob/master/src/core/geometry/modules/cad/CadTraverse.ts#L25 )

- 2. convert each edge to a BufferGeometry: https://github.com/polygonjs/polygonjs/blob/master/src/core/geometry/modules/cad/toObject3D/CadEdge.ts#L48, where I use GCPnts_UniformAbscissa_3 to get the points based on a curve tolerance.

**Reply by koodistrom** [reply]
Thank you very much! This is tremendously helpful. 

ps. Polygonjs seems like really interesting project gonna have to take a better look at it later when I have time
